### PR TITLE
Bug 1219162 - Sorting bookmark records in each folder. r=yifan

### PR DESCRIPTION
**Pull request by weilonge**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a4c6de1c55ad6e35e1ae91f95cbf99c9bbdeffb0

File: tv_apps/browser/js/sync/db.js [hunk 1]
```diff
@@ -194,7 +195,29 @@ var SyncBrowserDB = {
     } else if (indexes.bmkUri) {
       this.db.getBookmarkByUri(indexes.bmkUri, callback);
     } else if (indexes.parentid) {
-      this.db.getBookmarkByParentId(indexes.parentid, callback);
+      this.db.getBookmark(indexes.parentid, folder => {
+        if (folder && folder.type === 'folder' && folder.children) {
+          this.db.getBookmarkByParentId(indexes.parentid, list => {
+            var children = folder.children, result = [];
```
Comment (begeeben) [inline tv_apps/browser/js/sync/db.js]: Please separate var definition in multiple lines.

File: tv_apps/browser/js/sync/db.js [hunk 2]
```diff
@@ -194,7 +195,29 @@ var SyncBrowserDB = {
...
+            for (var i = 0; i < children.length; i++) {
+              for (var j = 0; j < list.length; j++) {
+                var oriId = list[j].id, id = oriId;
```
Comment (begeeben) [inline tv_apps/browser/js/sync/db.js]: Please separate var definition in multiple lines.

File: tv_apps/browser/js/sync/db.js [hunk 3]
```diff
@@ -194,7 +195,29 @@ var SyncBrowserDB = {
...
+                // Append the reduntant underline to match the ID from Syncto.
+                if (oriId.length < FXSYNCID_LEN) {
+                  id += new Array(FXSYNCID_LEN + 1 - oriId.length).join('_');
```
Comment (begeeben) [inline tv_apps/browser/js/sync/db.js]: Can this pass jshint? ```new Array``` is not recommanded.

An additional thought is, according to http://stackoverflow.com/a/14343876, the ```new Array``` padding method is very slow (though probably not much difference in this case). Please use the method described in http://stackoverflow.com/a/24398129 .

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e56ede639d63019eff7ee05e79f321a78bb2d952

**Pull request merged**